client/src/main/java/org/opensearch/ml/client/MachineLearningNodeClient.java

```diff
@@ -169,4 +229,77 @@ private void validateMLInput(MLInput mlInput, boolean requireInput) {
         }
     }
 
+    private String getAction(Map<String, Object> arguments) {
+        return (String) arguments.get(ACTION);
+    }
+
+    private FunctionName getFunctionName(Map<String, Object> arguments) {
+        String algo = (String) arguments.get(ALGORITHM);
+        if (algo == null) {
+            throw new  IllegalArgumentException("The parameter algorithm is required.");
+        }
+        switch (algo.toLowerCase(Locale.ROOT)) {
+            case MLConstants.KMEANS:
+                return KMEANS;
+            case MLConstants.RCF:
+                return arguments.get(AD_TIME_FIELD) == null ?
+                        BATCH_RCF : FIT_RCF;
+            default:
+                throw new IllegalArgumentException(
+                        String.format("unsupported algorithm: %s.", algo));
+        }
+    }
+
+    private MLAlgoParams convertArgumentToMLParameter(Map<String, Object> arguments,
+                                                        FunctionName func) {
+        switch (func) {
```

Comment:
For adding new algorithms, we need to change this client part as well. How about moving this part inside plugin or ml-algorithm , so we keep client part stable.

Reply:
Good suggestion, moved to common part as client only depends on common part.